Source/WebCore/loader/CookieJar.cpp

```diff
@@ -132,15 +132,29 @@ void CookieJar::setCookies(Document& document, const URL& url, const String& coo
         ASSERT_NOT_REACHED();
 }
 
-bool CookieJar::cookiesEnabled(const Document&) const
+bool CookieJar::cookiesEnabled(const Document& document) const
 {
+    auto cookieURL = document.cookieURL();
+    if (cookieURL.isEmpty())
+        return false;
+
+    auto pageID = document.pageID();
+    std::optional<FrameIdentifier> frameID;
+    if (auto* frame = document.frame())
+        frameID = frame->loader().frameID();
+
     if (auto* session = m_storageSessionProvider->storageSession())
-        return session->cookieAcceptPolicy() != HTTPCookieAcceptPolicy::Never;
+        return session->cookiesEnabled(document.firstPartyForCookies(), cookieURL, frameID, pageID, shouldRelaxThirdPartyCookieBlocking(document));
 
     ASSERT_NOT_REACHED();
     return false;
 }
 
+void CookieJar::cookiesEnabledAsync(const Document&, CompletionHandler<void(bool)>&& completionHandler) const
+{
+    completionHandler(false);
```

Comment:
Why is it OK to always return false here? I think this is only used by WebKitLegacy but I am not entirely clear why we return false.

Reply:
This is also called in WebKit as we create place holder Document for SharedWorker (`NonRenderedPlaceholder`), and Document will try fetch the value anyways when cookieURL and firstPartyForCookies is set. 
cookieEnabled is only available in Navigator, not WorkerNavigator, so I made it return false.